src/comm.jl

```diff
@@ -190,3 +190,46 @@ end
 handle(() -> nothing, "junorc")
 
 isconnected() = sock ≠ nothing && isopen(sock)
+
+# communication utilities
+# -----------------------
+
+"""
+    @tojs ex
+
+Easy constructor of `Dict` or `NamedTuple`. The result is supposed
+ to be parsed into JSON and send to a client over [`sock`](@ref).
+
+## Examples
+
+`ex`:
+- will be transformed into `NamedTuple` with parentheses: `@tojs (k: v)` is `((k = v,))`
+- will be transformed into `Dict` with braces: `@tojs {k: v}` is `Dict(:k => v)`
+- will be transformed into `NamedTuple` without parentheses and brances: `@tojs k: v` is `((k = v,))`
+- supports multiple property specs: `@tojs {k1: v1, k2: v2, ...}` is `Dict(:k1 => v1, :k2 => v2)`
+- supports [JS-like shorthand property notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer): `@tojs sym` is `((sym = sym,))`
+- can accept `=` instead of `:` notation: `@tojs {k = v}` is `@tojs {k: v}`
+"""
+macro tojs(ex)
```

Comment:
Wouldn't it make sense for this to produce a JSON string straight away?

Reply:
yep, it would be fine. I don't think we want to use this macro when we want to create `Dict` after all.

Reply:
wait, but if we parse them into JSON string in `@tojs`, then [`json` call in `msg`](https://github.com/JunoLab/Atom.jl/blob/5b85f56366166fb41855979b59a5720067b6faae/src/comm.jl#L129) will produce duplicated strings and it will be really breaking.

Reply:
No, we can wrap it in a `JSONText`.

Reply:
ah I get, thanks for the suggestion

Reply:
So after all, I found that I don't want the macro to produce JSON string. There are lots of cases where I want to refer to its fields after we constructed those "will be passed to JS" objects (e.g. tests or code body of goto.jl etc).
So in other word, what I really wanted was an easy constructor of `NamedTuple`s ... And so there might be no need of this "JS-like" macro..